src/RelayBoardV3.cpp

```diff
@@ -339,12 +341,20 @@ void RelayBoardV3::handle(std::shared_ptr<const pilot::RelayBoardV3Data> value){
 
 	out->release_structure_state = value->release_structure_state;
 
-	// ToDo add LED states
+	// Iterate over the LED color map
+	for (const auto& pair: value->led_states) {
+		if (pair.first == led_color_e::RED) {
+			out->led_state.r = pair.second;
+		} else if (pair.second == led_color_e::GREEN) {
```

Comment:
Should be `pair.first`. Also in the next `if`.

Reply:
good catch.. thanks!